packages/material-ui/src/Dialog/Dialog.js

```diff
@@ -117,11 +116,23 @@ export const styles = theme => ({
  * Dialogs are overlaid modal paper based components with a backdrop.
  */
 class Dialog extends React.Component {
+  componentWillUnmount() {
+    this.clickTarget = null;
```

Comment:
I think I made a mistake assuming we should clean up here. After all the instance is destroyed after unmounting anyway, right? It shouldn't be referenced after unmounting anyway.

It should however be cleaned up earlier. Theoretically we should only hold a weak reference in a WeakSet but that would require a polyfill. I think freeing this in `handleBackDropClick` is sufficient.

Reply:
I know it's not the right use but could a workaround be to use a WeakMap?

Reply:
There is a [WeakRef](https://github.com/tc39/proposal-weakrefs) proposal at stage 2 currently. I didn't find any notes about using a WeakMap or WeakSet instead so I guess it's not quite the same.